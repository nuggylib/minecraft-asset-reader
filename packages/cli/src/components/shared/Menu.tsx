import React from "react"
import SelectInput from "ink-select-input"
import { Box, Text } from "ink"

export const Menu = (props: {
  title: string
  options: {
    label: string
    value: string
  }[]
  onSelectHandler: (option: { label: string; value: string }) => void
}) => (
  <>
    <Box marginLeft={1}>
      <Text bold color="blue">
        {props.title}
      </Text>
    </Box>
    <SelectInput items={props.options} onSelect={props.onSelectHandler} />
  </>
)
