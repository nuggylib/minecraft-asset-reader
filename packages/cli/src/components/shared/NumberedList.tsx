import React from "react"
import { Text, Box, Newline } from "ink"

type NumberedListItem = {
  text: string
  subItems?: {
    [key: string]: string
  }
}

export const NumberedList = (props: { items: NumberedListItem[] }) => (
  <>
    {props.items.map((item, index) => (
      <Box flexDirection="column" key={`item_${index}`}>
        <Box marginLeft={2}>
          <Text bold>
            {index + 1}. {item.text}
          </Text>
        </Box>
        {!!item.subItems
          ? Object.keys(item.subItems).map((subItem) => (
              <Box marginLeft={4} key={`item_${index}_sub-item_${subItem}`}>
                <Text>
                  {subItem}. {item.subItems![subItem]}
                </Text>
              </Box>
            ))
          : null}
        <Newline />
      </Box>
    ))}
  </>
)
