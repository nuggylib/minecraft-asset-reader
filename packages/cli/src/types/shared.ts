// Defines the Int type
export type Int = number & { __int__: void }

export const roundToInt = (num: number): Int => Math.round(num) as Int

// TODO: Implement this patter for all internal APIs; need to do a lot of cleanup and standardization
export type MutationResult = {
  success: boolean
  message?: string
  data?: any
}
