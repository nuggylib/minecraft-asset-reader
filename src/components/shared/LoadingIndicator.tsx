import React from "react"
import { Text } from "ink"
import Spinner from "ink-spinner"

export const LoadingIndicator = (props: { loadingText: string }) => (
  <Text>
    <Text color="blue">
      <Spinner type="dots" />
    </Text>
    {`${props.loadingText}`}
  </Text>
)
