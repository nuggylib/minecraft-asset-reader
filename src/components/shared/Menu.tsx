import React from "react"
import SelectInput from "ink-select-input"
import { Box, Text } from "ink"
import { OPTION_VALUE } from "../../CLIApp"

export const Menu = (props: {
  title: string
  options: {
    label: string
    value: string
  }[]
  onSelectHandler: (option: { label: string; value: string }) => void
}) => (
  <>
    <Box marginLeft={1}>
      <Text bold color="blue">
        {props.title}
      </Text>
    </Box>
    <SelectInput items={props.options} onSelect={props.onSelectHandler} />
  </>
)
