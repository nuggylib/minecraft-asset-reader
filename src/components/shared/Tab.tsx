import { Text } from "ink"
import { times } from "lodash"
import React from "react"

/**
 * Helper component to make it easier to implement "tabs" in the terminal
 * output. We use 2 spaces instead of `\t` simply because the spacing is far too wide,
 * otherwise
 * 
 * @param param0 
 * @returns 
 */
export const Tab = ({ count }: { count: number }) => <Text>{
        times(count, () => `  `)
    }</Text>