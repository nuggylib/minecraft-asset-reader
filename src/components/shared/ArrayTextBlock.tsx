import { Box, Text } from "ink"
import React from "react"
import { Tab } from "./Tab"

export const ArrayTextBlock = ({label, array }: {label: string, nestingDepth: number, array: string[]}) => {
    return (
        <>
            <Text><Tab count={3}/>{label}: {array.map((val, index) => {
                let str = val
                if (index !== (array.length - 1)){
                    str += `, `
                }
                return str
            })}</Text>
        </>
    )
}