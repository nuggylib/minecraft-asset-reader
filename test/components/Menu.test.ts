import renderer from "react-test-renderer"
import { Menu } from "../../src/components/shared/Menu"

it(`renders correctly`, () => {
  const component = renderer
    .create(
      Menu({
        title: `TEST`,
        options: [
          {
            label: `Test label`,
            value: `test_value`,
          },
        ],
        onSelectHandler: () => {},
      })
    )
    .toJSON()

  // https://jestjs.io/docs/snapshot-testing
  expect(component).toMatchSnapshot()
})
