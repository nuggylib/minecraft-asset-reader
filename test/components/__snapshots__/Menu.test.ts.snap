// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <ink-box
    style={
      Object {
        "flexDirection": "row",
        "flexGrow": 0,
        "flexShrink": 1,
        "marginBottom": 0,
        "marginLeft": 1,
        "marginRight": 0,
        "marginTop": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      }
    }
  >
    <ink-text
      internal_transform={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "flexGrow": 0,
          "flexShrink": 1,
          "textWrap": "wrap",
        }
      }
    >
      TEST
    </ink-text>
  </ink-box>,
  <ink-box
    style={
      Object {
        "flexDirection": "column",
        "flexGrow": 0,
        "flexShrink": 1,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      }
    }
  >
    <ink-box
      style={
        Object {
          "flexDirection": "row",
          "flexGrow": 0,
          "flexShrink": 1,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
    >
      <ink-box
        style={
          Object {
            "flexDirection": "row",
            "flexGrow": 0,
            "flexShrink": 1,
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 1,
            "marginTop": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          }
        }
      >
        <ink-text
          internal_transform={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "flexGrow": 0,
              "flexShrink": 1,
              "textWrap": "wrap",
            }
          }
        >
          ‚ùØ
        </ink-text>
      </ink-box>
      <ink-text
        internal_transform={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "flexGrow": 0,
            "flexShrink": 1,
            "textWrap": "wrap",
          }
        }
      >
        Test label
      </ink-text>
    </ink-box>
  </ink-box>,
]
`;
